File: Crio/StackAndQueue/3_BinaryTreeInorderTraversal/Solution.java
import java.util.*;
import crio.ds.Tree.TreeNode;

/*
// Definition of TreeNode
public class TreeNode {
    public long val;
    public TreeNode left;
    public TreeNode right;

    public TreeNode (long x) {
        val = x;
        left = null;
        right = null;
    }
}
*/
public class Solution {
  // CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
  public static void dfs(TreeNode vertex , List<Long> inOrder) {
    if (vertex == null) {
      return;
    }
    dfs(vertex.left , inOrder);
    inOrder.add(vertex.val);
    dfs(vertex.right , inOrder);
  }
  // CRIO_SOLUTION_END_MODULE_L1_PROBLEMS

  public List<Long> inorderTraversal(TreeNode root) {
    // CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
    List<Long> inOrder = new ArrayList<Long>();
    dfs(root , inOrder);
    return inOrder;
    // CRIO_SOLUTION_END_MODULE_L1_PROBLEMS
  }
}

File: Crio/StackAndQueue/3_BinaryTreeInorderTraversal/Solution.cpp
/*
// Definition for TreeNode
class TreeNode {
public:
    long val;
    TreeNode* left;
    TreeNode* right;
    TreeNode* next;
    TreeNode (long x) {
        val = x;
        left = NULL;
        right = NULL;
        next = NULL;
    }
*/
class BinaryTreeInorderTraversal {
public:
    //CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
    void dfs(TreeNode* vertex, vector<int>& inOrder) {
        if (vertex == NULL) {
            return;
        }
        dfs(vertex->left, inOrder);
        inOrder.push_back(vertex->val);
        dfs(vertex->right, inOrder);
    }
    // CRIO_SOLUTION_END_MODULE_L1_PROBLEMS

    vector<int> inorderTraversal(TreeNode* root) {
        //CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
        vector<int> inOrder;
        dfs(root, inOrder);
        return inOrder;
        // CRIO_SOLUTION_END_MODULE_L1_PROBLEMS
    }
};